crates/rolldown/src/runtime/runtime-base.js

```diff
@@ -13,9 +13,10 @@ var __commonJS = (cb, mod) => function () {
   return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports
 }
 var __commonJSMin = (cb, mod) => () => (mod || cb((mod = { exports: {} }).exports, mod), mod.exports)
+// Used to implement ESM exports both for "require()" and "import * as"
 var __export = (target, all) => {
   for (var name in all)
-    __defProp(target, name, { get: all[name], enumerable: true })
+    __defProp(target, name, typeof all[name] === 'function' ? { get: all[name], enumerable: true } : { value: all[name], enumerable: true })
```

Comment:
Would this eventually require `get` vs `value` decision from build time?
I'm simply thinking about a case like this, which causes getter definition.

```js
__export(exports, { App: App })
function App() {}
```

Reply:
https://rollupjs.org/repl/?version=4.17.1&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJy4lMkZhLmpzJyU1Q25pbXBvcnQlMjAqJTIwYXMlMjBmb28lMjBmcm9tJTIwJy4lMkZmb28uanMnJTVDbiU1Q25jb25zb2xlLmxvZyhmb28pJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCUyQyU3QiUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBsZXQlMjBhJTIwJTNEJTIwMSU1Q25leHBvcnQlMjBmdW5jdGlvbiUyMHRlc3QoKSUyMCU3QiU3RCU1Q24lNUNuZXhwb3J0JTIwZGVmYXVsdCUyMHMlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlMkMlMjJuYW1lJTIyJTNBJTIyZm9vLmpzJTIyJTdEJTJDJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMG5zJTIwZnJvbSUyMCcuJTJGZm9vLmpzJyU1Q25ucy5hJTIwJTNEJTIwMTAwMCUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJhLmpzJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiU3RCU3RA==, nice catch, for function definition rollup always use function assign.

Reply:
> Would this eventually require `get` vs `value` decision from build time? I'm simply thinking about a case like this, which causes getter definition.
> 
> ```js
> __export(exports, { App: App })
> function App() {}
> ```

Seems this needs to be done in compile time.

Reply:
It depends on if the value could be safely `copy` or need to hold it's reference. Since the situation could mixed, so I guess we can't decide it in the compile time. Unless we don't use `__export` helpr, but that's another thing.

https://rollupjs.org/repl/?version=4.17.1&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJy4lMkZhLmpzJyU1Q25pbXBvcnQlMjAqJTIwYXMlMjBmb28lMjBmcm9tJTIwJy4lMkZmb28uanMnJTVDbiU1Q25jb25zb2xlLmxvZyhmb28pJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCUyQyU3QiUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBsZXQlMjBhJTIwJTNEJTIwMSU1Q25leHBvcnQlMjBmdW5jdGlvbiUyMHRlc3QoKSUyMCU3QiU3RCU1Q24lNUNuYSUyMCUzRCUyMDElNUNuJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjBzJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTJDJTIybmFtZSUyMiUzQSUyMmZvby5qcyUyMiU3RCUyQyU3QiUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAqJTIwYXMlMjBucyUyMGZyb20lMjAnLiUyRmZvby5qcyclNUNubnMuYSUyMCUzRCUyMDEwMDAlM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlMkMlMjJuYW1lJTIyJTNBJTIyYS5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlN0QlN0Q=

Reply:
Oh, I get it, this is a bug.